docs/guides/user-containers/readme.md

```diff
@@ -262,16 +262,119 @@ allowed per workflow.
+to select the NNF nodes that run the containers.
+
+For the `jobdw` directive that is included above, the servers (i.e. NNF nodes) must also be assigned
+along with the computes.
+
 ## Running a Container Workflow
 
-TODO: Once the container exit strategy is complete, come back to this section to detail:
+Once the workflow is created, the WLM progresses it through the following states. This is a quick
+overview of the container-related behavior that occurs:
+
+- Proposal: Verify [storages](#container-storages) are provided according to the container profile.
+- Setup: If applicable, [request ports](#container-ports) from NnfPortManager.
+- DataIn: No container related activity.
+- PreRun: Appropriate `MPIJob` or `Job(s)` are created for the workflow. In turn, user containers
+are created and launched by Kubernetes. Containers are expected to start in this state.
+- PostRun: Once in PostRun, user containers are expected to complete (non-zero exit)
+successfully.
+- DataOut: No container related activity.
+- Teardown: Ports are released; `MPIJob` or `Job(s)` are deleted, which in turn deletes the user
+containers.
+
+The two main states of a container workflow (i.e. PreRun, PostRun) are discussed further in the
+following sections.
+
+### PreRun
+
+In PreRun, the containers are created and expected to start. Once the containers reach a
+non-initialization state (i.e. Running), the containers are considered to be started and the
+workflow can advance.
+
+By default, containers are expected to start within 60 seconds. If not, the workflow reports an
+Error that the containers cannot be started. This value is configurable via the
+`preRunTimeoutSeconds` field in the container profile.
+
+To summarize the PreRun behavior:
+
+- If the container starts successfully (running), transition to `Completed` status.
+- If the container fails to start, transition to the `Error` status.
+- If the container is initializing and has not started after `preRunTimeoutSeconds` seconds,
+terminate the container and transition to the `Error` status.
+
+#### Init Containers
+
+The NNF Software injects [Init
+Containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) into the container
+specification to perform initialization tasks. These containers must run to completion before the
+main container can start.
+
+These initialization tasks include:
 
-- PreRun Behavior
-- PostRun Behavior
-- Teardown
+- Ensuring the proper permissions (i.e. UID/GID) are available in the main container
+- For MPI jobs, ensuring the launcher pod can contact each worker pod via DNS
 
-This will mostly be a from a WLM/DWS Workflow integration perspective, but will touch on the lower
-level workings.
+### PreRun Completed
+
+Once PreRun has transitioned to `Completed` status, the user container is now running.
+Sequentially, the WLM should initiate applications on the compute nodes. Utilizing [container
```

Comment:
What do you mean by "sequentially" here? The user apps on the compute nodes should be able to start at the same time.

Reply:
I was making the assumption that Flux would wait for Completed before starting the application on the compute nodes. Maybe I should reword?

Reply:
I read it as "start the application on compute nodes one at a time." Maybe that's just me.

Reply:
I can see that. I'll reword this.